### PR TITLE
Adds binary recursive stein algorithm to the GCD class

### PR DESCRIPTION
**Pull request by kayoslab**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description

 
* Three different approaches for the GCD to be calculated
* Improves the LCM by passing the used GCD algorithm into the function.
* Find an easy GCD function if possible.
* Adopts the changes in the appropriate README file.

### CODE REVIEW

File: GCD/README.markdown [hunk 1]
```diff
@@ -88,9 +88,9 @@ The `max()` and `min()` at the top of the function make sure we always divide th
 
 An idea related to the GCD is the *least common multiple* or LCM.
 
-The least common multiple of two numbers `a` and `b` is the smallest positive integer that is a multiple of both. In other words, the LCM is evenly divisible by `a` and `b`. 
+The least common multiple of two numbers `a` and `b` is the smallest positive integer that is a multiple of both. In other words, the LCM is evenly divisible by `a` and `b`. The example implementation of the LCM takes two numbers and a specification which GCD algorithm is used.
 
-For example: `lcm(2, 3) = 6` because 6 can be divided by 2 and also by 3.
+For example: `lcm(2, 3, using: gcdIterativeEuklid) = 6` because 6 can be divided by 2 and also by 3.
```
Comment (richard-ash) [inline GCD/README.markdown]: Keep this as `lcm(2, 3) = 6`

File: GCD/GCD.swift [hunk 2]
```diff
@@ -1,34 +1,124 @@
...
+ - Parameter m: First natural number.
+ - Parameter n: Second natural number.
+ - Parameter using: The used gcd algorithm to calculate the lcm.
+ - Throws: Can throw a `divisionByZero` error if one of the given
+ attributes turns out to be zero or less.
+ - Returns: The least common multiplier of the two attributes as
+ an unsigned integer
+ */
+func lcm(_ m: Int, _ n: Int, using gcdAlgorithm: (Int, Int) -> (Int)) throws -> Int {
+    guard (m & n) != 0 else { throw LCMError.divisionByZero }
```
Comment (richard-ash) [inline GCD/GCD.swift]: nit: You don't need parentheses around `m & n`

**Comment by kayoslab** [comment]
@richard-ash Thank you for the valid comments. I updated my pull request according to that. The example code is now moved into the playground's `Sources` folder and the README is extended with details regarding all three algorithms and a short discussion.

**Comment by richard-ash** [comment]
Awesome thanks so much @kayoslab. Changes look 💯

**Pull request merged**